app/models/input_template_renderer.rb

```diff
@@ -0,0 +1,39 @@
+class InputTemplateRenderer
+  include Foreman::Renderer
+
+  attr_accessor :template, :host, :invocation, :error_message
+
+  def initialize(template, host = nil, invocation = nil)
+    @host = host
+    @invocation = invocation
+    @template = template
+  end
+
+  def render
+    render_safe(@template.template, [ :input ])
+  rescue => e
+    self.error_message = _('error during rendering: %s') % e.message if error_message.nil?
+    Rails.logger.debug e.to_s + "\n" + e.backtrace.join("\n")
```

Comment:
Not sure I follow, what logger should be used? We don't have plugin specific logger so we're using Rails global logger. We can't call `logger`, if that's what you mean, since we are in pure Ruby object.